opencog/atomspace/Handle.h

```diff
@@ -71,13 +73,33 @@ friend class AtomspaceHTabler;    // persistance
     AtomPtr resolve_ptr();
     static void set_resolver(const AtomTable*);
     static void clear_resolver(const AtomTable*);
+
+    // Set the UUID in a handle. This is the safe version. Use  this version
+    // if you have any doubts at all about the possibility that the handle
+    // may have been resolved. This sets the new UUID and also calls 
+    // resets the shared_ptr AtomPtr so it doesn't point to anything,
+    // the reference is decremented, and the object deleted if this is the
+    // last one. This is used by AtomTable to avoid having to construct
+    // temporaries while iterating when the iteration itself can resolve the
+    // AtomPtr in order to clear the AtomPtr reference so it won't be pointing
+    // to an atom that has a UUID that is different than the UUID passed
+    // to set_uuid.    
+    void set_uuid(const UUID uuid) { _ptr.reset(); _uuid = uuid; }
```

Comment:
It is not going backwards, it is surfacing the backwards that was hidden because someone changed the interface to use Handles at the higher levels while still using UUIDs in the lower internals.